pkg/util/strategicpatch/patch.go

```diff
@@ -31,7 +31,352 @@ import (
 // lists should be merged or replaced.
 //
 // For more information, see the PATCH section of docs/api-conventions.md.
+
+// CreateStrategicMergePatch creates a patch that can be passed to StrategicMergePatch.
```

Comment:
We'll need 2 diff modes:

One that detects all changes: changes, deletions, additions. Used to generate the patch, and maybe to determine whether anything has changed.

One that detects just changes and deletions. Additions would be ignored. This would be used to detect divergence. Initially we could barf if divergence were detected. Eventually we'll want --force to override the changes in the live state. 

Export (#13038) could be the means by which a new config could be generated from the live state, if that's what the user wanted.